tapscript/tx.go

```diff
@@ -165,10 +167,25 @@ func virtualTxOut(txAsset *asset.Asset) (*wire.TxOut, error) {
 	}
 	outputTree := mssmt.NewCompactedTree(mssmt.NewDefaultStore())
 
-	// TODO(bhandras): thread the context through.
-	tree, err := outputTree.Insert(context.TODO(), key, leaf)
-	if err != nil {
-		return nil, err
+	var (
+		tree mssmt.Tree
+	)
+
+	switch sighashType {
+	case txscript.SigHashAll:
+		// TODO(bhandras): thread the context through.
+		tree, err = outputTree.Insert(context.TODO(), key, leaf)
+		if err != nil {
+			return nil, err
+		}
+
+	case txscript.SigHashNone:
+		nullKey := [32]byte{}
+		nullLeaf := mssmt.LeafNode{}
+		tree, err = outputTree.Insert(context.TODO(), nullKey, &nullLeaf)
```

Comment:
I think instead of inserting a null key/leaf, we just skip inserting anything? Leading to an empty tree instead? Since the null key is never really looked up anyway.